src/integration.test.ts

```diff
@@ -0,0 +1,7000 @@
+import {
+  createLiteral,
+  createIdentifier,
+  createVariableDeclaration,
+} from '@src/lang/create'
+import type {
+  ChamferParameters,
+  EdgeTreatmentParameters,
+  FilletParameters,
+} from '@src/lang/modifyAst/addEdgeTreatment'
+import {
+  EdgeTreatmentType,
+  deleteEdgeTreatment,
+  getPathToExtrudeForSegmentSelection,
+  hasValidEdgeTreatmentSelection,
+  modifyAstWithEdgeTreatmentAndTag,
+} from '@src/lang/modifyAst/addEdgeTreatment'
+import { getEdgeCutMeta, getNodeFromPath } from '@src/lang/queryAst'
+import { getNodePathFromSourceRange } from '@src/lang/queryAstNodePathUtils'
+import { codeRefFromRange } from '@src/lang/std/artifactGraph'
+import { topLevelRange } from '@src/lang/util'
+import { isOverlap } from '@src/lib/utils'
+import { addSubtract } from '@src/lang/modifyAst/boolean'
+import type { NodePath } from '@rust/kcl-lib/bindings/NodePath'
+import type { Operation } from '@rust/kcl-lib/bindings/Operation'
+import { defaultSourceRange } from '@src/lang/sourceRange'
+import { filterOperations, getOperationVariableName } from '@src/lib/operations'
+import {
+  getCalculatedKclExpressionValue,
+  getStringValue,
+} from '@src/lib/kclHelpers'
+import { getSafeInsertIndex } from '@src/lang/queryAst/getSafeInsertIndex'
+import type { Coords2d } from '@src/lang/std/sketch'
+import { isPointsCCW } from '@src/lang/wasm'
+import { closestPointOnRay } from '@src/lib/utils2d'
+import { expect } from 'vitest'
+import type { VariableDeclaration } from '@src/lang/wasm'
+import { updateCenterRectangleSketch } from '@src/lib/rectangleTool'
+import { trap } from '@src/lib/trap'
+import fs from 'node:fs'
+import { ARG_INDEX_FIELD, LABELED_ARG_FIELD } from '@src/lang/queryAstConstants'
+import type { Parameter } from '@src/lang/wasm'
+import { modelingMachine } from '@src/machines/modelingMachine'
+import { createActor } from 'xstate'
+import { vi } from 'vitest'
+import { getConstraintInfoKw } from '@src/lang/std/sketch'
+import { ARG_END_ABSOLUTE, ARG_INTERIOR_ABSOLUTE } from '@src/lang/constants'
+import { removeSingleConstraintInfo } from '@src/lang/modifyAst'
+import { getTagDeclaratorsInProgram } from '@src/lang/queryAst/getTagDeclaratorsInProgram'
+import { modelingMachineDefaultContext } from '@src/machines/modelingSharedContext'
+
+// Unfortunately, we need the real engine here it seems to get sweep faces populated
+beforeAll(async () => {
+  await initPromise
+  await new Promise((resolve) => {
+    engineCommandManager
+      .start({
+        token: env().VITE_KITTYCAD_API_TOKEN || '',
+        width: 256,
+        height: 256,
+        setStreamIsReady: () => {
+          console.log('no op for a unit test')
+        },
+        callbackOnUnitTestingConnection: () => {
+          resolve(true)
+        },
+      })
+      .catch(reportRejection)
+  })
+}, 30_000)
+
+afterAll(() => {
+  engineCommandManager.tearDown()
+})
+
+// TODO: two different methods for the same thing. Why?
```

Comment:
Heyyyy those were just quick not-repeating-4-lines-of-code helpers that weren't exactly the same across mods as the needs were different, you can put my name of up for cleaning that up it was always the plan but not priority cause it's just test code. Sorry that caused you extra trouble!